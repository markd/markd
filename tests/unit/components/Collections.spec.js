import { shallowMount, mount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/Collections.vue'

import Vuex from 'vuex'
import VueRouter from 'vue-router'

const localVue = createLocalVue()

localVue.use(Vuex)
localVue.use(VueRouter)

const router = new VueRouter()

const collectionOne = 'Web Development'
const collectionTwo = 'Spicy Memes'

const store = new Vuex.Store({
  state: {
    pinned: [
      {
        icon: 'cloud_queue',
        name: 'All Bookmarks'
      },
      {
        icon: 'favorite_border',
        name: 'Favourite Bookmarks'
      }
    ],
    collections: [
      {
        colour: 'red',
        name: collectionOne
      },
      {
        colour: 'green',
        name: collectionTwo
      }
    ]
  },
  getters: {
    getCollections: state => {
      return state.collections
    }
  }
})

describe('Collections.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Component, {
      localVue,
      store
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('has proper childrens', () => {
    const wrapper = mount(Component, {
      localVue,
      store,
      router
    })

    const names = wrapper.findAll('.sidebar__groupItemName')

    expect(names.length).toBe(2)
    expect(names.at(0).text()).toBe(collectionOne)
    expect(names.at(1).text()).toBe(collectionTwo)
  })
})
