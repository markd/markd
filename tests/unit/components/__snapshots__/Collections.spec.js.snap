// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections.vue renders 1`] = `
<div class="sidebar__group collections">
  <div class="sidebar__groupItems">
    <sidebaritem-stub route="collection" collection="[object Object]"></sidebaritem-stub>
    <sidebaritem-stub route="collection" collection="[object Object]"></sidebaritem-stub>
  </div>
</div>
`;
